ui/job-view/pushes/FuzzyJobFinder.jsx

```diff
@@ -0,0 +1,293 @@
+              ? fuse.search(this.state.fuzzySearch)
+              : this.props.jobList,
+          });
+        });
+      }
+    } else {
+      this.setState({
+        fuzzyList: this.props.jobList,
+      });
+    }
+  };
+
+  resetForm = () => {
+    this.setState({
+      selectedList: [],
+    });
+  };
+
+  addAllJobs = () => {
+    const addJobsRef = this.addJobsRef.current;
+    const selectedOptions = Array.from(
+      addJobsRef.options,
+      option => option.textContent,
+    );
+    let { selectedList } = this.state;
+
+    // When adding jobs, add only new, unique job names to avoid duplicates
+    selectedList = [...new Set([].concat(selectedList, selectedOptions))];
+    this.setState({ selectedList });
+    addJobsRef.selectedIndex = -1;
+  };
+
+  removeAllJobs = () => {
+    this.setState({
+      selectedList: [],
+    });
+  };
+
+  addJobs = () => {
+    const addJobsRef = this.addJobsRef.current;
+    const selectedOptions = Array.from(
+      addJobsRef.selectedOptions,
+      option => option.textContent,
+    );
+    let { selectedList } = this.state;
+
+    // When adding jobs, add only new, unique job names to avoid duplicates
+    selectedList = [...new Set([].concat(selectedList, selectedOptions))];
+    this.setState({ selectedList });
+    addJobsRef.selectedIndex = -1;
+  };
+
+  removeJobs = () => {
+    const removeJobsRef = this.removeJobsRef.current;
+    const selectedOptions = Array.from(
+      removeJobsRef.selectedOptions,
+      option => option.textContent,
+    );
+    let { selectedList } = this.state;
+
+    selectedList = selectedList.filter(
+      value => !selectedOptions.includes(value),
+    );
+    this.setState({ selectedList });
+    removeJobsRef.selectedIndex = -1;
+  };
+
+  submitJobs = () => {
+    const { notify } = this.props;
+    if (this.state.selectedList.length > 0) {
+      notify('Submitting selected jobs...');
+      this.setState({
+        submitDisabled: true,
+      });
+      PushModel.triggerNewJobs(
+        this.state.selectedList,
+        this.props.decisionTaskId,
+      )
+        .then(result => {
+          notify(result, 'success');
+          this.props.toggle();
+        })
+        .catch(e => {
+          notify(formatTaskclusterError(e), 'danger', { sticky: true });
+          this.setState({
+            submitDisabled: false,
+          });
+        });
+    } else {
+      notify('Please select at least one job from the list', 'danger');
+    }
+  };
+
+  toggle = () => {
+    this.setState({
+      modal: !this.state.modal,
+    });
+  };
+
+  updateAddButton = () => {
+    this.setState({
+      addDisabled: this.addJobsRef.current.selectedIndex === -1,
+    });
+  };
+
+  updateRemoveButton = () => {
+    this.setState({
+      removeDisabled: this.removeJobsRef.current.selectedIndex === -1,
+    });
+  };
+
+  // Implement an alphanumeric sort so mochitest-10 is sorted after mochitest 9, not mochitest-1
+  sortAlphaNum = (a, b) => {
```

Comment:
I think it might be worth moving this under `ui/helpers/`

Reply:
Done.